src/imap.rs

```diff
@@ -2375,23 +2375,23 @@ async fn add_all_recipients_as_contacts(
             .map(|s| normalize_name(s))
             .unwrap_or_default();
 
-        match Contact::add_or_lookup(
-            context,
-            &display_name_normalized,
-            &recipient.addr,
-            Origin::OutgoingTo,
-        )
-        .await?
-        {
-            Some((_, modified)) => {
+        match ContactAddress::new(&recipient.addr) {
+            Err(err) => warn!(
+                context,
+                "Could not add contact for recipient with address {:?}: {:#}", recipient.addr, err
+            ),
+            Ok(recipient_addr) => {
+                let (_, modified) = Contact::add_or_lookup(
```

Comment:
Up2u but maybe declare a var for contact addr instead? Highly indented code is harder to read